### PR TITLE
Enable Github Auth for production

### PR DESCRIPTION
**Pull request by victorb**
WIP, right now it's only applying config for dev

### CODE REVIEW

**Comment by victorb** [comment]
Gonna make this use patch files to apply the production configuration, instead of templating.

**Comment by victorb** [comment]
@lgierth @Kubuxu ready for review!

File: README.md [hunk 1]
```diff
@@ -41,3 +41,11 @@ New hosts need:
 
 Secrets are only valid for dev-environment. Make sure to only run jenkins in dev
 listening to 127.0.0.1 and not 0.0.0.0. Secrets are replaced when jenkins is deployed
+
```
Comment (Kubuxu) [inline README.md:44]: README is outdated

Comment (victorb) [reply README.md:44]: Fixed! Thanks!

**Review by Kubuxu — APPROVED** [review]

File: build-prod.sh [hunk 2]
```diff
@@ -4,6 +4,9 @@ set -e
 
 IMAGE_NAME="quay.io/ipfs/jenkins"
 
+# Making sure jenkins-secrets are not included in the image
+git submodule deinit .
```
Comment (ghost) [inline build-prod.sh]: Add `-f` here -- "If --force is specified, the submodule’s work tree will be removed even if it contains local modifications." We care mostly about the work tree, and not so much about the submodule config.

**Comment by ghost** [comment]
Great, LGTM, one comment

File: build-prod.sh (hunk 2 above)
Comment (victorb) [reply build-prod.sh]: Very good catch! Fixed

**Comment by victorb** [comment]
@lgierth addressed! Thanks! Merging this one

**Pull request merged**